pkg/crc/machine/kubeconfig.go

```diff
@@ -181,13 +180,10 @@ func getTokenForUser(username, password, ip string, ca []byte, clusterConfig *ty
 			},
 			DialContext: func(ctx gocontext.Context, network, address string) (net.Conn, error) {
 				logging.Debugf("Using address: %s", address)
-				hostname, port, err := net.SplitHostPort(address)
+				_, port, err := net.SplitHostPort(address)
 				if err != nil {
 					return nil, err
 				}
-				if strings.HasSuffix(hostname, constants.AppsDomain) {
-					port = strconv.Itoa(int(ingressHTTPSPort))
-				}
```

Comment:
@vyasgun After removing it looks like it fails again, did you test this scenario?
```
DEBU Using address: api.crc.testing:6443          
DEBU Dialing to 127.0.0.1:6443                    
DEBU Using address: oauth-openshift.apps-crc.testing:443 
DEBU Dialing to 127.0.0.1:443                     
ERRO Cannot update kubeconfig: dial tcp 127.0.0.1:443: connect: connection refused 
DEBU Making call to close driver server           
```

Reply:
I didn't face the issue. I will take another look and update.

Reply:
I added the lines back and updated the PR.